src/main/java/org/simplify4u/plugins/sign/openpgp/PGPKeyInfo.java

```diff
@@ -50,20 +50,31 @@
     @Builder
     private PGPKeyInfo(String keyId, String keyPass, File keyFile) {
 
-        id = Optional.ofNullable(Optional.ofNullable(System.getenv(SIGN_KEY_ID_ENV)).orElse(keyId))
+        id = Optional.ofNullable(stringFromEnv(SIGN_KEY_ID_ENV).orElse(keyId))
                 .map(PGPKeyInfo::parseKeyId)
                 .orElse(null);
 
-        pass = Optional.ofNullable(Optional.ofNullable(System.getenv(SIGN_KEY_PASS_ENV)).orElse(keyPass))
+        pass = Optional.ofNullable(stringFromEnv(SIGN_KEY_PASS_ENV).orElse(keyPass))
                 .map(String::toCharArray)
                 .orElse(null);
 
-        key = Optional.ofNullable(System.getenv(SIGN_KEY_ENV))
+        key = stringFromEnv(SIGN_KEY_ENV)
                 .map(String::trim)
                 .map(PGPKeyInfo::keyFromString)
                 .orElseGet(() -> keyFromFile(keyFile));
     }
 
+    /**
+     * Read environment variable and filter by "null" string - this value is set be invoker-maven-plugin.
+     *
+     * @param environmentName a environment variable name
+     * @return content of environment variable or empty if not exist.
+     */
+    private static Optional<String> stringFromEnv(String environmentName) {
+        return Optional.ofNullable(System.getenv(environmentName))
```

Comment:
That is a really scary hack, as it has  nothing to do with our code, but solely with a bug in the invoker plugin. Shouldn't we better fix the maven invoker plugin instead?

Reply:
You right, I will prepare issue for maven-invoker-plugin.

Reply:
Great! If you post the link to the PR here, I will be happy review it. :-)

Reply:
PR ready https://github.com/apache/maven-shared-utils/pull/68
and issues:
https://issues.apache.org/jira/browse/MSHARED-969
https://issues.apache.org/jira/browse/MINVOKER-273

but way is long to fix it by Apache ... accept PR, new release of maven-shared-utils, update maven-shared-utils in maven-invoker-plugin ... new release of maven-invoker-plugin

so we need this workaround

Reply:
If you need the fix right now, certainly that way is long. But do we actually? I have no problem with adopting the workaround, I just think in this case we should open an issue as a reminder so we do not forget to remove the workaround once the final solution is publicly available.

Reply:
I added TODO - sonar will report this as issue so we will remember